storage/SMAIndexSubBlock.cpp

```diff
@@ -0,0 +1,661 @@
+/**
+ *   Copyright 2016 Pivotal Software, Inc.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ **/
+
+#include "storage/SMAIndexSubBlock.hpp"
+
+#include <algorithm>
+#include <cstdint>
+#include <cstring>
+#include <iterator>
+#include <utility>
+#include <exception>
+
+#include "catalog/CatalogAttribute.hpp"
+#include "catalog/CatalogRelationSchema.hpp"
+#include "compression/CompressionDictionary.hpp"
+#include "expressions/predicate/ComparisonPredicate.hpp"
+#include "expressions/predicate/Predicate.hpp"
+#include "expressions/predicate/PredicateCost.hpp"
+#include "expressions/scalar/Scalar.hpp"
+#include "expressions/scalar/ScalarAttribute.hpp"
+#include "storage/CompressedTupleStorageSubBlock.hpp"
+#include "storage/StorageBlockLayout.pb.h"
+#include "storage/StorageConstants.hpp"
+#include "storage/StorageErrors.hpp"
+#include "storage/SubBlockTypeRegistry.hpp"
+#include "storage/TupleIdSequence.hpp"
+#include "storage/TupleStorageSubBlock.hpp"
+#include "types/Type.hpp"
+#include "types/TypeFactory.hpp"
+#include "types/TypeID.hpp"
+#include "types/TypedValue.hpp"
+#include "types/operations/binary_operations/BinaryOperation.hpp"
+#include "types/operations/binary_operations/BinaryOperationFactory.hpp"
+#include "types/operations/binary_operations/BinaryOperationID.hpp"
+#include "types/operations/comparisons/Comparison.hpp"
+#include "types/operations/comparisons/ComparisonFactory.hpp"
+#include "types/operations/comparisons/ComparisonID.hpp"
+#include "types/operations/comparisons/ComparisonUtil.hpp"
+#include "utility/BitVector.hpp"
+#include "utility/Macros.hpp"
+#include "utility/PtrVector.hpp"
+#include "utility/ScopedBuffer.hpp"
+
+#include "glog/logging.h"
+
+using std::memcpy;
+using std::memmove;
+using std::pair;
+using std::size_t;
+using std::sort;
+using std::uint8_t;
+using std::uint16_t;
+using std::uint32_t;
+using std::vector;
+using std::exception;
+
+namespace quickstep {
+
+QUICKSTEP_REGISTER_INDEX(SMAIndexSubBlock, SMA);
+
+typedef sma_internal::EntryReference EntryReference;
+typedef sma_internal::SMAPredicate SMAPredicate;
+typedef sma_internal::SMAHeader SMAHeader;
+typedef sma_internal::SMAEntry SMAEntry;
+typedef sma_internal::Selectivity Selectivity;
+
+namespace sma_internal {
+
+/**
+ * @return Selectivity of an equals predicate.
+ */
+inline Selectivity getSelectivity_E(const TypedValue &literal,
```

Comment:
You can ignore my comment, I didn't realize I was looking at a cpp file.

In general, a ``static`` function can be called on the class itself, without creating an object. In your case, I don't think ``static`` makes sense.